From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: allolive <160342668+allolive@users.noreply.github.com>
Date: Tue, 29 Sep 2026 11:59:27 -0400
Subject: [PATCH 01/14] power menu: reboot, suspend and shutdown only

Trims DialogButtonMenu to Reboot, Suspend and Shutdown in that order,
dropping Exit, the shutdown timers, Hibernate, eMMC reboot, Restart Kodi,
Logoff, master mode and inhibit-idle. New power icon.
---
 xml/DialogButtonMenu.xml | 90 ++++++++--------------------------------
 1 file changed, 18 insertions(+), 72 deletions(-)

diff --git a/xml/DialogButtonMenu.xml b/xml/DialogButtonMenu.xml
index 3323896..fd31717 100644
--- a/xml/DialogButtonMenu.xml
+++ b/xml/DialogButtonMenu.xml
@@ -16,78 +16,24 @@
 			</include>
 			<control type="panel" id="9000">
 				<include>ButtonMenuList</include>
-				<content>
-					<item>
-						<label>$LOCALIZE[13012]</label>
-						<onclick>Quit()</onclick>
-						<visible>System.ShowExitButton</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13016]</label>
-						<onclick>Powerdown()</onclick>
-						<visible>System.CanPowerDown</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[20150]</label>
-						<onclick>AlarmClock(shutdowntimer,Shutdown())</onclick>
-						<visible>!System.HasAlarm(shutdowntimer)</visible>
-						<visible>System.CanPowerDown | System.CanSuspend | System.CanHibernate</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[20151] $INFO[System.AlarmPos,(,)]</label>
-						<onclick>CancelAlarm(shutdowntimer)</onclick>
-						<visible>System.HasAlarm(shutdowntimer)</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13011]</label>
-						<onclick>Dialog.Close(shutdownmenu)</onclick>
-						<onclick>Suspend()</onclick>
-						<visible>System.CanSuspend</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13010]</label>
-						<onclick>Dialog.Close(shutdownmenu)</onclick>
-						<onclick>Hibernate()</onclick>
-						<visible>System.CanHibernate</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13013]</label>
-						<onclick>Reset()</onclick>
-						<visible>System.CanReboot</visible>
-					</item>
-					<item>
-						<label>Reboot from eMMC/NAND</label>
-						<onclick>System.ExecWait("/usr/sbin/rebootfromnand")</onclick>
-						<onclick>Reset()</onclick>
-						<visible>System.PathExist("/dev/system") | System.PathExist("/dev/super") | System.PathExist("/dev/userdata") | System.PathExist("/dev/data")</visible>
-					</item>
-					<item>
-						<label>Restart Kodi</label>
-						<onclick>RestartApp()</onclick>
-					</item>
-					<item>
-						<label>$LOCALIZE[20126] $INFO[system.profilename]</label>
-						<onclick>dialog.close(all,true)</onclick>
-						<onclick>System.LogOff</onclick>
-						<visible>System.HasLoginScreen | Integer.IsGreater(System.ProfileCount,1)</visible>
-						<visible>System.Loggedon</visible>
-					</item>
-					<item>
-						<label>$VAR[MasterModeLabel]</label>
-						<onclick>mastermode</onclick>
-						<visible>System.HasLocks</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13017]</label>
-						<onclick>InhibitIdleShutdown(true)</onclick>
-						<visible>System.HasShutdown +!System.IdleShutdownInhibited</visible>
-					</item>
-					<item>
-						<label>$LOCALIZE[13018]</label>
-						<onclick>InhibitIdleShutdown(false)</onclick>
-						<visible>System.HasShutdown + System.IdleShutdownInhibited</visible>
-					</item>
-				</content>
+					<content>
+						<item>
+							<label>$LOCALIZE[13013]</label>
+							<onclick>Reset()</onclick>
+							<visible>System.CanReboot</visible>
+						</item>
+						<item>
+							<label>$LOCALIZE[13011]</label>
+							<onclick>Dialog.Close(shutdownmenu)</onclick>
+							<onclick>Suspend()</onclick>
+							<visible>System.CanSuspend</visible>
+						</item>
+						<item>
+							<label>$LOCALIZE[13016]</label>
+							<onclick>Powerdown()</onclick>
+							<visible>System.CanPowerDown</visible>
+						</item>
+					</content>
 			</control>
 		</control>
 	</controls>
